revocation/ocsp/ocsp.go

```diff
@@ -167,17 +167,25 @@ func extensionsToMap(extensions []pkix.Extension) map[string][]byte {
 func executeOCSPCheck(cert, issuer *x509.Certificate, server string, opts Options) (*ocsp.Response, error) {
 	// TODO: Look into other alternatives for specifying the Hash
 	// https://github.com/notaryproject/notation-core-go/issues/139
-	ocspRequest, err := ocsp.CreateRequest(cert, issuer, &ocsp.RequestOptions{Hash: crypto.SHA256})
+	// The following do not support SHA256 hashes:
+	//  - Microsoft
+	//  - Entrust
+	//  - Let's Encrypt
+	//  - Digicert (sometimes)
+	// As this represents a large percentage of public CAs, we are using the
+	// hashing algorithm SHA1, which has been confirmed to be supported by all
+	// that were tested.
+	ocspRequest, err := ocsp.CreateRequest(cert, issuer, &ocsp.RequestOptions{Hash: crypto.SHA1})
 	if err != nil {
 		return nil, GenericError{Err: err}
 	}
 
 	var resp *http.Response
-	if base64.URLEncoding.EncodedLen(len(ocspRequest)) >= 255 {
+	if base64.StdEncoding.EncodedLen(len(ocspRequest)) >= 255 {
 		reader := bytes.NewReader(ocspRequest)
 		resp, err = opts.HTTPClient.Post(server, "application/ocsp-request", reader)
 	} else {
-		encodedReq := base64.URLEncoding.EncodeToString(ocspRequest)
+		encodedReq := url.QueryEscape(base64.StdEncoding.EncodeToString(ocspRequest))
```

Comment:
`encodedReq` is passed as a path instead of query. I think it would be `PathEscape` instead of `QueryEscape`.

Reference: RFC 2560:

> GET {url}/{url-encoding of base-64 encoding of the DER encoding of
   the OCSPRequest}

Reply:
After testing, I can confirm that PathEscape will not work. My guess is that it is a result of URL encoding the "=" characters at the end of base64. We need to use QueryEscape. I have also confirmed that not escaping will also work, so there is not a risk related to lacking URL encoding characters.

Reply:
Resolving since we cannot change this function